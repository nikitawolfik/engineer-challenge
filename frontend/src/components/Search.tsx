import React from "react";

interface Props {
  value: string;
  onChange: (e: React.ChangeEvent<HTMLInputElement>) => void;
  label: string;
  name: string;
}

const Search = ({ value, onChange, label, name }: Props) => {
  return (
    <div className="flex flex-col">
      <label className="text-xs ml-2 mb-1" htmlFor={`input-${name}`}>
        {label}
      </label>
      <input
        className="rounded-lg p-2 border-solid border-slate-50 border-2"
        name={name}
        value={value}
        placeholder={label}
        onChange={onChange}
      />
    </div>
  );
};

export default Search;
