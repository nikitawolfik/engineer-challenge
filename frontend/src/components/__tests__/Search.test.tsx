/**
 * @jest-environment jsdom
 */

import React from "react";
import { render } from "@testing-library/react";
import "@testing-library/jest-dom";
import Search from "components/Search";

describe("Search input", () => {
  test("Snapshot", async () => {
    const { container } = render(
      <Search
        label="Hello label"
        name="Test name"
        value=""
        onChange={jest.fn()}
      />
    );

    expect(container).toMatchSnapshot();
  });
});
