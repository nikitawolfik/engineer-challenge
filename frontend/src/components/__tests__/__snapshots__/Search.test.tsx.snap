// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search input Snapshot 1`] = `
<div>
  <div
    class="flex flex-col mb-2"
  >
    <label
      class="text-xs ml-2 mb-1"
      for="input-Test name"
    >
      Hello label
    </label>
    <input
      class="rounded-lg p-2 border-solid border-slate-50 border-2"
      name="Test name"
      placeholder="Hello label"
      value=""
    />
  </div>
</div>
`;
